### PR TITLE
Python 3 Conversion - Group 6

### PR DESCRIPTION
**Pull request by DavidFair**
Description of work.

As part of #16739 this converts all the python algorithms in group 6. 
Additionally I've changed the print statements to use the logger instead - however I can revert this if needed. 

---

Following this comment here:
#16739 (comment)

I have also removed TOFTOFConvertToDeltaE - I can restore it if this isn't the correct place to do it?

---

**To test:**
All unit tests for the Python Interface should pass

_Does not need to be in the release notes._

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by DavidFair** [comment]
Following this comment here:
https://github.com/mantidproject/mantid/issues/16739#issuecomment-234180120

I have also removed TOFTOFConvertToDeltaE - I can restore it if this isn't the correct place to do it?

**Comment by rosswhitfield** [comment]
`VesuvioTOFFit` and `VesuvioCorrections.py` both still fail to load in mantid when using python3 because they use `scripts/Inelastic/vesuvio/backgrounds.py` and `scripts/Inelastic/vesuvio/profiles.py`. I suggest also doing the python3 conversion of these files in this pull request.

```
Failed to load plugin /home/rwp/mantid/Framework/PythonInterface/plugins/algorithms/VesuvioTOFFit.py. Error: invalid syntax (backgrounds.py, line 77)
Failed to load plugin /home/rwp/mantid/Framework/PythonInterface/plugins/algorithms/VesuvioCorrections.py. Error: invalid syntax (backgrounds.py, line 77)
Failed to load plugin /home/rwp/mantid/Framework/PythonInterface/plugins/algorithms/VesuvioTOFFit.py. Error: invalid syntax (profiles.py, line 296)
Failed to load plugin /home/rwp/mantid/Framework/PythonInterface/plugins/algorithms/VesuvioCorrections.py. Error: invalid syntax (profiles.py, line 296)
```

**Comment by DavidFair** [comment]
Thanks for the heads up - I suspect my copy of Mantid still build against 2.7 giving me the impression that these scripts worked.

**Comment by rosswhitfield** [comment]
Code looks good. Can you look at the related unit tests for these algorithms. These ones fail with python3:
```
	580 - PythonAlgorithms_SuggestTibHYSPECTest (Failed)
	582 - PythonAlgorithms_SymmetriseTest (Failed)
	588 - PythonAlgorithms_TOFTOFMergeRunsTest (Failed)
	598 - PythonAlgorithms_VesuvioPreFitTest (Failed)
	599 - PythonAlgorithms_VesuvioResolutionTest (Failed)
	600 - PythonAlgorithms_VesuvioThicknessTest (Failed)
	601 - PythonAlgorithms_VesuvioTOFFitTest (Failed)
```

**Comment by DavidFair** [comment]
There was an issue with how `simpleAPI.py` handled both positional and keyword arguments. 
`set_properties` relies on the positional arguments being set before keyword arguments, however because dictionaries are not ordered in Python 3 we cannot rely on it getting the order correct so instead we keep the different argument types separate.

File: scripts/Inelastic/vesuvio/commands.py [hunk 1]
```diff
@@ -305,7 +307,7 @@ def _update_masses_from_params(old_masses, param_ws):
     @return The modified mass dictionary
     """
     for mass in old_masses:
-        for param in mass.keys():
+        for param in list(mass.keys()):
```
Comment (rosswhitfield) [inline scripts/Inelastic/vesuvio/commands.py]: Don't need to convert to list here

**Comment by rosswhitfield** [comment]
You missed `VesuvioPreFitTest.py`

**Comment by DavidFair** [comment]
Should be all fixed all tests pass locally and some issues with tests randomly failing should be resolved however if you still get any randomly fail could you paste a stack trace as there are some things in Vesuvio which rely on dictionaries being ordered.

**Comment by rosswhitfield** [comment]
LGTM :shipit:

**Pull request merged**